Space MP5 - Redux

Offwrldr opened the pull request:
# [ORIGINAL PR HERE](https://github.com/Bubberstation/Bubberstation/pull/2194)

## About The Pull Request

Recently, a fellow by the name of Cato Christopher made me aware of a conversion kit within the game that is unimplemented. From what I could see, it was possibly never completed - the RomTech SMG, a conversion of the Sindano to make it fully-automatic. The magazine sprite, for example, was incorrectly aligned with the original sprite. There was talk in the code about making it researchable. I have taken it in as my own.

Introducing the _Shaytan_.

![Sprite-000ddd2](https://github.com/user-attachments/assets/c6b9b8d9-c476-4d48-a95d-0c6778f21796)

Description: A Romulus-produced conversion kit. This kit comes with a modified sear to enable seamless full-automatic fire - at the cost of fire-select function. This kit includes an alloy and polymer body, a proprietary **integrated flashlight**, an **alternate handguard and modified barrel for the addition of universal suppressors**. Accepts any standard Sol pistol magazine.

It has been tested. It passed. See performance [here.](https://github.com/Bubberstation/Bubberstation/pull/2194#issuecomment-2406524916)

![image_2024-10-10_184048524](https://github.com/user-attachments/assets/33b4dbef-7f96-4fc3-b4d8-4d57ee48832e)

BONUS ROUND: I modified my own dono sprite. It works. It is added to the MultiSec wardrobes. It has identical stats to baseline Sec armors.

EDIT:
The original collared vest:
![image](https://github.com/user-attachments/assets/df595ac2-0ac9-4c81-985e-c5f4b17931ca)
The new collared vest:
![image](https://github.com/user-attachments/assets/edadb403-febc-4974-8c4f-feab2f4a981b)
![image](https://github.com/user-attachments/assets/ebc6e19b-7260-41de-8fbb-cf17532bfb50)



## Why It's Good For The Game

It adds diversity to the Sec weapons pool. To prevent abuse, I made it pricey to order from Cargo. It also does not show up on the Armory menu (I think) of Sec's Cargo orders - meaning they have to GO to Cargo to order it. The fire-rate can also be adjusted to make it work better for gameplay reasons.

It is intended to be a mid-round option for Security Officers.

## Proof Of Testing

![image](https://github.com/user-attachments/assets/66e4d1b6-21f7-40f8-afad-26dccd5b59f7)
![image](https://github.com/user-attachments/assets/66b4a21b-6184-4359-8739-9c8b5c25e45c)


## Changelog
:cl: Bangle
add: Added new mechanics or gameplay changes - added Shaytan SMG conversion kit to Cargo. Added the BNGL Collared Vest to MultiSec - it has identical stats and I don't wanna lock it behind a donowall.
add: Added more things - added Shaytan SMG conversion kit
image: added/modified/removed some icons or images - changed a bunch of the sprites, including my own Collared Vest
spellcheck: fixed a few typos - changed descriptions and renamed my Collared Vest. I think it's the BNGL Collared Vest now.
code: changed some code - see above.
/:cl:

Offwrldr commented:
## NOTICE: THE ORIGINAL PR WAS CLOSED DUE TO MULTIPLE COMMITS BUILDING UPON EACH OTHER.

To preserve the discussion and history there, please click [here](https://github.com/Bubberstation/Bubberstation/pull/2194) to read the original PR.

Majkl-J reviewed (changes requested):
I feel like the sprite could see less of a hard outline, most sprites we have use ~1 darkness level above on the pallete for edges instead of a static outline of one color

Paxilmaniac reviewed (changes requested):
![image](https://github.com/user-attachments/assets/f20ee51f-2c0c-463b-ae80-8642e4a69707)

Majkl-J commented:
Staled out, reopen if work continues

StrangeWeirdKitten commented:
Bangle DM'd me

Offwrldr commented:
![image](https://github.com/user-attachments/assets/d16f1dc3-1011-4e85-9946-03d833eda7f2)
Spritened Lightened

Offwrldr commented:
This will be my final commit to this PR unless it's for bugfixing or some other such thing. Not sure why the checks aren't passing and Github makes my head hurt.

### VEST CHANGES

I believe my current dono item, the BNGL Collared Vest, to look too protective for what it does. I also should not have named it the BNGL, because that is silly and I'm not quite sure what I was thinking. As such:

1. I have modified the sprite so that it looks less protective.
2. I have changed the name.
3. I have created reskins for Sec.

Provided below are the changes.

Neutral
![image](https://github.com/user-attachments/assets/33e952f1-0d36-4f8b-9fb8-ebd7f88c3ba5)

Red
![image](https://github.com/user-attachments/assets/25b9396e-2815-46b9-be03-a1a6fd8d6458)

Blue
![image](https://github.com/user-attachments/assets/a1356ecc-75e3-4c6d-9275-3d64758359f4)

Updated name: "GLP-C 'Úlfur' Vest"
Updated desc: "A set of General Light Protective armor, with complimentary pauldrons and an additional armored collar, similar to a gorget. This pattern of armored vest is typically afforded to diplomats and members of the press in hostile environments, though has seen use among private security forces. The armored collar is designed to protect the neck and throat from shrapnel. All things considered, it's quite comfortable, though many unfortunate wearers are often mistaken for combatants given the militarized aesthetic."

Majkl-J commented on modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm:
```diff
@@ -126,8 +126,21 @@
 	item_type = /obj/item/gun/ballistic/automatic/sol_rifle/marksman
 	cost = PAYCHECK_COMMAND * 8
 
+ShaytanAndVest
+//These are new - added by Bangle. Framework for additional conversion kits as deemed appropriate by maintainers.
+/datum/armament_entry/company_import/sol_defense/conversionkit
+	restricted = TRUE
+	subcategory = "Conversion Kits"
+
+/datum/armament_entry/company_import/sol_defense/conversionkit/sindano
+	item_type = /obj/item/weaponcrafting/gunkit/sol_smg_rapidfire_kit
+	cost = PAYCHECK_COMMAND * 8
+
+/*
+=======
 
 
+master
```
Traces of failed conflict resolution. Probably why checks are failing

Majkl-J commented on modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm:
```diff
@@ -125,6 +125,30 @@
+	item_type = /obj/item/weaponcrafting/gunkit/sol_smg_rapidfire_kit
+	cost = PAYCHECK_COMMAND * 8
+
+/*
+/datum/armament_entry/company_import/sol_defense/longarm/renoster
+	item_type = /obj/item/gun/ballistic/shotgun/riot/sol
+	cost = PAYCHECK_COMMAND * 6
+
+/datum/armament_entry/company_import/sol_defense/longarm/sindano
+	item_type = /obj/item/gun/ballistic/automatic/sol_smg
+	cost = PAYCHECK_COMMAND * 6
+
+/datum/armament_entry/company_import/sol_defense/longarm/elite
+	item_type = /obj/item/gun/ballistic/automatic/sol_rifle/marksman
+	cost = PAYCHECK_COMMAND * 8
```
```suggestion
/datum/armament_entry/company_import/sol_defense/longarm/renoster
	item_type = /obj/item/gun/ballistic/shotgun/riot/sol
	cost = PAYCHECK_COMMAND * 6

/datum/armament_entry/company_import/sol_defense/longarm/sindano
	item_type = /obj/item/gun/ballistic/automatic/sol_smg
	cost = PAYCHECK_COMMAND * 6

/datum/armament_entry/company_import/sol_defense/longarm/elite
	item_type = /obj/item/gun/ballistic/automatic/sol_rifle/marksman
	cost = PAYCHECK_COMMAND * 8

/*
```
